### PR TITLE
Auto detect cilium incompatibility during cni loop

### PR DESCRIPTION
**Pull request by howardjohn**
This is a UX improvement for users. Note this is not just logging, but also limiting to 1 loop per second. This ensures when we do trigger the condition, we don't spin at max CPU and log 100s of times per second

### CODE REVIEW

File: cni/pkg/install/install.go [hunk 1]
```diff
@@ -304,3 +311,51 @@ func setNotReady(isReady *atomic.Value) {
 	installReady.Record(0)
 	isReady.Store(false)
 }
+
```
Comment (bleggett) [inline cni/pkg/install/install.go:314]: This LGTM. Once the next `libcni` release is cut and the container runtimes get it, we should get subdirectory-based loading "for free" and will no longer have conflicts anyway, and can probably nuke most of this watching/rewriting junk.

**Review by bleggett — APPROVED** [review]

**Comment by bleggett** [comment]
> This is a UX improvement for users. Note this is not just logging, but also limiting to 1 loop per second. This ensures when we do trigger the condition, we don't spin at max CPU and log 100s of times per second

Thinking about this some more - the downside of this is that at this point your cluster is already broken, so  slowing the loop really just creates bigger windows for your cluster to be broken-er.

It would be better ultimately to have `istio-cni` go unhealthy if there are X detected rewrites in X seconds I think, there is no self-healing possible here so backoffs in and of themselves don't accomplish much.

**Comment by howardjohn** [comment]
It should already be "unhealthy" when this occurs, or at least was in my experience

**Comment by bleggett** [comment]
> It should already be "unhealthy" when this occurs, or at least was in my experience

Good ty for confirming - I thought as much but couldn't remember.

**Pull request merged**